packages/shared/lib/services/sync/sync.service.ts

```diff
@@ -249,75 +249,57 @@ export const getSyncs = async (
     });
 };
 
```

Comment:
below consolidating `getSyncsByProviderConfigKey` and `getSyncsByConnectionId` by adding a filter argument instead of having a bunch of different function doing more or less the same thing